src/Aspire.Cli/Commands/RunCommand.cs

```diff
@@ -614,4 +620,81 @@ private async Task<bool> MonitorProcessesForTerminationAsync(AppHostInformation
         // Timeout reached
         return false;
     }
+
+    private async Task EnsureUserSecretsInitializedAsync(FileInfo projectFile, CancellationToken cancellationToken)
+    {
+        try
+        {
+            // Check if the project file has a UserSecretsId element
+            var projectContent = await File.ReadAllTextAsync(projectFile.FullName, cancellationToken).ConfigureAwait(false);
+            
+            // Use XDocument to properly parse the XML and check for UserSecretsId
+            var doc = System.Xml.Linq.XDocument.Parse(projectContent);
+            var hasUserSecretsId = doc.Descendants().Any(e => e.Name.LocalName == "UserSecretsId");
+            
+            // If the project already has a UserSecretsId, we don't need to do anything
+            if (hasUserSecretsId)
+            {
+                return;
+            }
+
+            // Run 'dotnet user-secrets init' to add a UserSecretsId to the project
+            _logger.LogDebug("Initializing user secrets for project {ProjectFile}", projectFile.FullName);
+            
+            var startInfo = new ProcessStartInfo("dotnet")
+            {
+                WorkingDirectory = projectFile.Directory?.FullName ?? Directory.GetCurrentDirectory(),
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+            
+            startInfo.ArgumentList.Add("user-secrets");
+            startInfo.ArgumentList.Add("init");
+            startInfo.ArgumentList.Add("--project");
+            startInfo.ArgumentList.Add(projectFile.FullName);
```

Comment:
Should this happen automatically vs giving the user a prompt that they need to do it? It feels a little automagical that aspire run automatically modifies source code.

Reply:
Agree. I'll undo.